Match enet_tasklet_disconnect() declaration to definition

elm3 opened the pull request:
### Description

The declaration of enet_tasklet_disconnect() was declared in the header to return a void.  In definition, it returned int8_t.  This fix is just to correct the header to match the code.

### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

elm3 committed:
Match enet_tasklet_disconnect() declaration to definition 

The declaration of enet_tasklet_disconnect() was declared in the header to return a void.  In definition, it returned int8_t.  This fix is just to correct the header to match the code.

elm3 committed:
Use signature of enet_tasklet_XXXX() calls in code.

The declaration of enet_tasklet_XXXX() functions was declared in the header file.  Some of the declarations didn't match the
definitions so those were fixed.  This let to some other issues, where the calls where not called cleanly.  All this was
corrected and is now building cleanly.

kjbracey commented:
As you've now noted in the issue - the parameters also mismatch, so fix that too.

Also, it seems the error was able to arise because ethernet_tasklet.c failed to include "include/enet_tasklet.h"

Please fix that at the same time - the other tasklets have their own-header include before "include/mesh_system.h".

kjbracey commented:
The `send_cb` parameter should be `true` in this case.

That does raise the question of what happened to any calls that pass `false`. It appears such calls used to happen in various destructors, but they vanished in commit 27cced74. Intentionally, @SeppoTakalo ?

elm3 committed:
fixed the signature enet_tasklet_disconnect()

In the earlier commit, I fixed the return code.  In this fix, I added the arguments so that the signature exactly matches the function.

elm3 committed:
include enet_tasklet header in enet_tasklet source

By including this header in the source, we are having the compiler keep the function and declaration the same.

SeppoTakalo reviewed (changes requested):
Build failures now.

kjbracey commented:
As I said above, the one remaining call to this needs to pass `true`. The calls passing `false` from destructors no longer exist.

elm3 committed:
enet_tasklet_disconnect called with arg where req.

the enet_tasklet_disconnect() signature is now fixed to correctly require an argument (boolean).

This call (the only call) now calls the function with a boolean correctly.

elm3 committed:
fixed previous call to pass the CORRECT argument

I added the wrong boolean in this fixed call.

elm3 committed:
Fixing the enet_tasklet_network_init() signature

The signature said it returned uint8_t but in reality the function returned int8_t.

elm3 committed:
Included header with types defined.

"mesh_connection_status_t" is not defined without the header.

SeppoTakalo reviewed (approved):
Looks good, but please squash the commits together.

7 commits and 5 lines changed :D

elm3 committed:
Merge branch 'elm3-patch-1' of https://github.com/ARMmbed/mbed-os into elm3-patch-1

cmonr commented:
@elm3 Could you take a look at your last commit? Something doesn't look right with it.

Whenever you do a rebase, you end up force-pushing your branch, not generating a merge commit.

cmonr commented:
@elm3 ?

cmonr commented:
Closing since no response/activity has occurred in over two weeks.

Feel free to reopen once an update is available.
